### PR TITLE
piece model, game,piece,user associations

### PR DESCRIPTION
**Pull request by yuzu-r**
piece model, children models, game-user-piece assocations.

### CODE REVIEW

**Comment by jonianven** [comment]
The model/associations look good and make sense 👍  I am wondering if we'd prefer to have subclasses (queen, bishop, etc.) in one file a la https://samurails.com/tips/sti-keeping-subclasses-file/.

**Comment by yuzu-r** [comment]
Yeah, I wasn't sure how many methods each type would end up having in the end and thought it might get too cluttered.

**Comment by jonianven** [comment]
Good point... I'm merging 🚢

File: spec/factories/kings.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FactoryGirl.define do
```
Comment (gruzzlymug) [inline spec/factories/kings.rb:1]: You might consider putting the piece factories into a single file

File: db/schema.rb [hunk 2]
```diff
@@ -11,19 +11,34 @@
...
-    t.integer  "user_id"
+    t.datetime "created_at",      null: false
+    t.datetime "updated_at",      null: false
+    t.integer  "white_player_id"
+    t.integer  "black_player_id"
+    t.string   "status"
   end
 
-  add_index "games", ["user_id"], name: "index_games_on_user_id", using: :btree
+  add_index "games", ["white_player_id"], name: "index_games_on_white_player_id", using: :btree
```
Comment (gruzzlymug) [inline db/schema.rb:28]: Consider adding an index on the black player id too

**Pull request merged**